esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -40,6 +40,80 @@ static const char* ESDL_CHANGE_PATH="/ESDL/Subscription/Change";
 
 extern bool trimXPathToParentSDSElement(const char *element, const char * xpath, StringBuffer & parentNodeXPath);
 
+class CEsdlDefinitionInfo : implements IEsdlDefinitionInfo, public CInterface
+{
+    friend class CEsdlSDSStore;
+
+private:
+    typedef MapStringTo<StringArray*> MapStringToStringArray;
+
+    MapStringToStringArray m_serviceMethodMap;
+    StringArray m_services;
+    Owned<IProperties> m_metadata;
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    CEsdlDefinitionInfo()
+    {
+        m_metadata.setown(createProperties());
+    }
+
+    ~CEsdlDefinitionInfo()
+    {
+        HashIterator iter(m_serviceMethodMap);
+        for(iter.first(); iter.isValid(); iter.next())
+        {
+            StringArray* arr = *m_serviceMethodMap.mapToValue(&iter.query());
+            delete arr;
+        }
+    }
+
+    virtual IProperties* queryMetadata() override
+    {
+        return m_metadata;
+    }
+
+    virtual StringArray* queryServices() override
+    {
+        return &m_services;
+    }
+
+    virtual StringArray* queryMethods(const char* service) override
+    {
+        StringArray** methods = m_serviceMethodMap.getValue(service);
+        if( methods )
+            return *(methods);
+        else
+            return NULL;
+    }
+
+private:
+    void addMetadata(const char* name, const char* value)
+    {
+        m_metadata->setProp(name, value);
+    }
+
+    void addService(const char* service)
+    {
+        if( m_services.contains(service) == false )
+            m_services.append(service);
+        if( m_serviceMethodMap.find(service) == NULL )
+            m_serviceMethodMap.setValue(service, new StringArray);
+    }
+
+    void addMethodToService(const char* service, const char* method)
+    {
+        if( m_serviceMethodMap.find(service) == NULL )
+        {
+            this->addService(service);
+        }
+
+        StringArray* methods = *m_serviceMethodMap.getValue(service);
+        methods->append(method);
```

Comment:
In short it will handle that case without error by adding a duplicate entry to the array. 

Even though it would be ill-advised to do so, it is possible to publish a definition containing two methods with the same name in one service. No errors are thrown, and this code will list both of them through the 'GetESDLDefinition' method.

Reply:
I think we should consider it an error and probably check things like that when the ESDL is published.  I really don't like the idea of two methods with the same name.

Reply:
Ok I'll open a ticket to add that error check (on publication only- not also here).